server/status_test.go

```diff
@@ -294,10 +294,28 @@ func startServerAndGetStatus(t *testing.T, keyPrefix string) (*TestServer, []byt
 		t.Fatal(err)
 	}
 
-	// Make sure the node is spun up and that a full scan of the ranges in the
-	// stores is complete.  The best way to do that is to wait twice.
-	ts.node.waitForScanCompletion()
-	ts.node.waitForScanCompletion()
+	// Make sure the range and spun up with an arbitrary read command. We do not
+	// expect a response.
+	if _, err := ts.db.Get("a"); err != nil {
+		t.Fatal(err)
+	}
+
+	// Make sure the node status is available. This is done by forcing stores to
+	// publish their status, synchronizing to the event feed with a canary
+	// event, and then forcing the server to write summaries immediately.
+	if err := ts.node.publishStoreStatuses(); err != nil {
+		t.Fatalf("error publishing store statuses: %v", err)
+	}
+	syncEvent := status.NewTestSyncEvent(1)
+	ts.EventFeed().Publish(syncEvent)
+	select {
+	case <-syncEvent.Sync:
+	case <-time.After(5 * time.Second):
+		t.Fatal("syncFeed() failed to complete before timeout 5 seconds")
```

Comment:
Moved the synchronization bit, with timeout, to a `Sync(timeout)` function on TestSyncEvent. This is fixed as part of that.